core/src/main/java/com/netflix/iceberg/util/ThreadPools.java

```diff
@@ -27,42 +27,21 @@
 import java.util.concurrent.ThreadPoolExecutor;
 
 public class ThreadPools {
-  public static final String PLANNER_THREAD_POOL_SIZE_PROP =
-      SystemProperties.PLANNER_THREAD_POOL_SIZE_PROP;
   public static final String WORKER_THREAD_POOL_SIZE_PROP =
       SystemProperties.WORKER_THREAD_POOL_SIZE_PROP;
 
-  private static ExecutorService PLANNER_POOL = MoreExecutors.getExitingExecutorService(
-      (ThreadPoolExecutor) Executors.newFixedThreadPool(
-          getPoolSize(PLANNER_THREAD_POOL_SIZE_PROP, 4),
-          new ThreadFactoryBuilder()
-              .setDaemon(true)
-              .setNameFormat("iceberg-planner-pool-%d")
-              .build()));
+  public static final int WORKER_THREAD_POOL_SIZE = getPoolSize(
+      WORKER_THREAD_POOL_SIZE_PROP,
+      Runtime.getRuntime().availableProcessors());
 
   private static ExecutorService WORKER_POOL = MoreExecutors.getExitingExecutorService(
       (ThreadPoolExecutor) Executors.newFixedThreadPool(
-          getPoolSize(WORKER_THREAD_POOL_SIZE_PROP, Runtime.getRuntime().availableProcessors()),
+          WORKER_THREAD_POOL_SIZE,
           new ThreadFactoryBuilder()
               .setDaemon(true)
```

Comment:
I think `MoreExecutors.getExitingExecutorService ` already creates daemon threads

Reply:
This passes a `ThreadFactoryBuilder` to set the name format. You're right that `getExitingExecutorService` should use daemon threads, but I don't think that is a reason to omit it here since that is a requirement of this code.

Reply:
That makes sense!